Maintenance/511 requirement levels (#195)
ConsumerDataStandardsAustralia/standards-staging: slate/source/includes/secondary/_energy_apis.md

The following variations to the endpoints published by AEMO from the energy sector endpoints apply:

* The `x-fapi-auth-date` header **MUST NOT** be passed to AEMO and AEMO **MUST NOT** require this header
* The `x-fapi-customer-ip-address` header **MUST NOT** be passed to AEMO and AEMO **MUST NOT** require this header
* The `x-cds-client-headers` header **MUST NOT** be passed to AEMO and AEMO **MUST NOT** require this header

> * A new header named `x-cds-arrangement` **MUST** be passed to AEMO for every invocation. This header **SHOULD** contain the arrangement ID for the consent that the request is being made under and will be used for tracing and audit purposes. This field **MUST** be populated but AEMO **MUST NOT** seek to validate the consent associated with the arrangement

Reviewer: Restating the same thing is bad form.

```suggestion
* A new header named `x-cds-arrangement` **MUST** be passed to AEMO for every invocation. This header **SHOULD** contain the arrangement ID for the consent that the request is being made under and will be used for tracing and audit purposes. AEMO **MUST NOT** seek to validate the consent associated with the arrangement
```

* Fields in the links object for all responses **MUST** be translated by the Data Holder into
values that are valid for a Data Recipient to be able to call back to the Data Holder
* The *Get Service Points* end point **MUST** be changed from a GET to a POST and will have the same request payload as the *Get Usage For Specific Service Points* endpoint

#### Additional Requirements

The following statements also apply to the endpoints published by AEMO:

* General headers **SHOULD** be provided as if the request were coming from the primary Data Holder and not propagated from the call made by the Data Recipient
* The `x-fapi-interaction-id` header **MUST** be propagated from the Data Recipient call to AEMO to allow for end to end tracing. If not supplied by the Data Recipient, the primary Data Holder **MUST** create a unique value for the `x-fapi-interaction-id` header before calling AEMO
* Endpoints that require knowledge of the NMIs that belong to the CDR Consumer have been

> excluded from the AEMO endpoint set. This includes *Get Bulk Usage* and *Get Bulk DER*.  When a primary Data Holder is required to respond to these endpoints they **SHOULD** call the equivalent
> endpoint for specific service points and provide the specific list of NMIs to AEMO

Reviewer: Isn't this a MUST? There's no other way and it opens the door to calling individual NMIs?

```suggestion
excluded from the AEMO endpoint set. This includes *Get Bulk Usage* and *Get Bulk DER*.  When a primary Data Holder is required to respond to these endpoints they **MUST** call the *Get Usage For Specific Service Points (SR)* and  *Get DER For Specific Service Points (SR)* respectively.
```
